compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java

```diff
@@ -201,24 +203,30 @@ public void save() {
         Path buildFilePath = this.sourceRoot.resolve(TARGET_DIR_NAME).resolve(BUILD_FILE);
         boolean shouldUpdate = this.currentPackage().getResolution().autoUpdate();
         // if build file does not exists
+
         if (!buildFilePath.toFile().exists()) {
-            // set both last build time and lat updated time as current timestamp
             createBuildFile(buildFilePath);
             writeBuildFile(buildFilePath);
             writeDependencies();
         } else {
-            // build file exists
-            BuildJson buildJson = readBuildJson(buildFilePath);
+            BuildJson buildJson = null;
+            try {
+                buildJson = readBuildJson(buildFilePath);
+            } catch (JsonSyntaxException e) {
+                // ignore
+            } catch (IOException e) {
+                return;
```

Comment:
Returning here will stop writing the Dependencies.toml. We should not do that. We should ignore both exceptions and continue.